src/main/java/impl/image/ConvolutionProviderImp.java

```diff
@@ -0,0 +1,47 @@
+package impl.image;
+
+import api.image.ConvolutionProvider;
+
+import java.awt.*;
+
+public class ConvolutionProviderImp implements ConvolutionProvider {
+
+    public Color[][] apply(Color[][] image, double[][] kernel) {
+        int n = image.length;
+        int m = image[0].length;
+        int kernelN = kernel.length;
+        int kernelM = kernel[0].length;
+        int HalfKernelN = kernelN / 2;
+        int HalfKernelM = kernelM / 2;
+        Color[][] img = new Color[n][m];
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < m; j++) {
+                int blue = 0;
+                int red = 0;
+                int green = 0;
+                int index1 = 0;
```

Comment:
It is still not clear for me: why do you need these indices and what do they mean? It is better to use more clear wording